### PR TITLE
chore(fetch-cache): remove zod from fetch cache

### PR DESCRIPTION
**Pull request by huozhi**
## Why?

Revert zod validation due to it increasing the next-runtime size.

Closes NEXT-3234

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 712f71a1611b95dfa49bccad9ee0277cf8253778

`pnpm test-start test/e2e/app-dir/app-external/app-external.test.ts`
- app dir - external dependency > should be able to opt-out 3rd party packages being bundled in server components

<details>
<summary>Expand output</summary>

  ● app dir - external dependency › should be able to opt-out 3rd party packages being bundled in server components

    expect(received).toContain(expected) // indexOf

    Expected substring: "opt-out-react-version: 18.2.0"
    Received string:    "<!DOCTYPE html><html><head><meta charSet=\"utf-8\"/><meta name=\"viewport\" content=\"width=device-width, initial-scale=1\"/><link rel=\"preload\" as=\"script\" fetchPriority=\"low\" href=\"/_next/static/chunks/webpack-d160adaf6d1d4b11.js\"/><script src=\"/_next/static/chunks/a5c3caef-b404cfa10a7b1d36.js\" async=\"\"></script><script src=\"/_next/static/chunks/472-627e0d90f650e085.js\" async=\"\"></script><script src=\"/_next/static/chunks/main-app-e47335c31f2da62b.js\" async=\"\"></script><script src=\"/_next/static/chunks/app/react-server/optout/page-bf733ad946da72ef.js\" async=\"\"></script><meta name=\"next-size-adjust\"/><script src=\"/_next/static/chunks/polyfills-78c92fac7aa8fdd8.js\" noModule=\"\"></script></head><body><div>Server: index.default<br/>Server subpath: subpath.default<br/>Client: index.default<br/>Client subpath: subpath.default<br/><p id=\"optout-react-version\">opt-out-react-version: 18.3.0</p></div><script src=\"/_next/static/chunks/webpack-d160adaf6d1d4b11.js\" async=\"\"></script><script>(self.__next_f=self.__next_f||[]).push([0]);self.__next_f.push([2,null])</script><script>self.__next_f.push([1,\"1:I[5127,[],\\\"\\\"]\\n3:I[2393,[\\\"605\\\",\\\"static/chunks/app/react-server/optout/page-bf733ad946da72ef.js\\\"],\\\"default\\\"]\\n4:I[7901,[],\\\"\\\"]\\n5:I[9154,[],\\\"\\\"]\\n7:I[5405,[],\\\"\\\"]\\n8:[]\\n\"])</script><script>self.__next_f.push([1,\"0:[null,[\\\"$\\\",\\\"$L1\\\",null,{\\\"buildId\\\":\\\"TwkIoZxfFJ0qFk1-IIdCk\\\",\\\"assetPrefix\\\":\\\"\\\",\\\"initialCanonicalUrl\\\":\\\"/react-server/optout\\\",\\\"initialTree\\\":[\\\"\\\",{\\\"children\\\":[\\\"react-server\\\",{\\\"children\\\":[\\\"optout\\\",{\\\"children\\\":[\\\"__PAGE__\\\",{}]}]}]},\\\"$undefined\\\",\\\"$undefined\\\",true],\\\"initialSeedData\\\":[\\\"\\\",{\\\"children\\\":[\\\"react-server\\\",{\\\"children\\\":[\\\"optout\\\",{\\\"children\\\":[\\\"__PAGE__\\\",{},[[\\\"$L2\\\",[\\\"$\\\",\\\"div\\\",null,{\\\"children\\\":[\\\"Server: index.default\\\",[\\\"$\\\",\\\"br\\\",null,{}],\\\"Server subpath: subpath.default\\\",[\\\"$\\\",\\\"br\\\",null,{}],[\\\"$\\\",\\\"$L3\\\",null,{}],[\\\"$\\\",\\\"p\\\",null,{\\\"id\\\":\\\"optout-react-version\\\",\\\"children\\\":\\\"opt-out-react-version: 18.3.0\\\"}]]}]],null],null]},[\\\"$\\\",\\\"$L4\\\",null,{\\\"parallelRouterKey\\\":\\\"children\\\",\\\"segmentPath\\\":[\\\"children\\\",\\\"react-server\\\",\\\"children\\\",\\\"optout\\\",\\\"children\\\"],\\\"error\\\":\\\"$undefined\\\",\\\"errorStyles\\\":\\\"$undefined\\\",\\\"errorScripts\\\":\\\"$undefined\\\",\\\"template\\\":[\\\"$\\\",\\\"$L5\\\",null,{}],\\\"templateStyles\\\":\\\"$undefined\\\",\\\"templateScripts\\\":\\\"$undefined\\\",\\\"notFound\\\":\\\"$undefined\\\",\\\"notFoundStyles\\\":\\\"$undefined\\\",\\\"styles\\\":null}],null]},[\\\"$\\\",\\\"$L4\\\",null,{\\\"parallelRouterKey\\\":\\\"children\\\",\\\"segmentPath\\\":[\\\"children\\\",\\\"react-server\\\",\\\"children\\\"],\\\"error\\\":\\\"$undefined\\\",\\\"errorStyles\\\":\\\"$undefined\\\",\\\"errorScripts\\\":\\\"$undefined\\\",\\\"template\\\":[\\\"$\\\",\\\"$L5\\\",null,{}],\\\"templateStyles\\\":\\\"$undefined\\\",\\\"templateScripts\\\":\\\"$undefined\\\",\\\"notFound\\\":\\\"$undefined\\\",\\\"notFoundStyles\\\":\\\"$undefined\\\",\\\"styles\\\":null}],null]},[[\\\"$\\\",\\\"html\\\",null,{\\\"children\\\":[[\\\"$\\\",\\\"head\\\",null,{}],[\\\"$\\\",\\\"body\\\",null,{\\\"children\\\":[\\\"$\\\",\\\"$L4\\\",null,{\\\"parallelRouterKey\\\":\\\"children\\\",\\\"segmentPath\\\":[\\\"children\\\"],\\\"error\\\":\\\"$undefined\\\",\\\"errorStyles\\\":\\\"$undefined\\\",\\\"errorScripts\\\":\\\"$undefined\\\",\\\"template\\\":[\\\"$\\\",\\\"$L5\\\",null,{}],\\\"templateStyles\\\":\\\"$undefined\\\",\\\"templateScripts\\\":\\\"$undefined\\\",\\\"notFound\\\":[[\\\"$\\\",\\\"title\\\",null,{\\\"children\\\":\\\"404: This page could not be found.\\\"}],[\\\"$\\\",\\\"div\\\",null,{\\\"style\\\":{\\\"fontFamily\\\":\\\"system-ui,\\\\\\\"Segoe UI\\\\\\\",Roboto,Helvetica,Arial,sans-serif,\\\\\\\"Apple Color Emoji\\\\\\\",\\\\\\\"Segoe UI Emoji\\\\\\\"\\\",\\\"height\\\":\\\"100vh\\\",\\\"textAlign\\\":\\\"center\\\",\\\"display\\\":\\\"flex\\\",\\\"flexDirection\\\":\\\"column\\\",\\\"alignItems\\\":\\\"center\\\",\\\"justifyContent\\\":\\\"center\\\"},\\\"children\\\":[\\\"$\\\",\\\"div\\\",null,{\\\"children\\\":[[\\\"$\\\",\\\"style\\\",null,{\\\"dangerouslySetInnerHTML\\\":{\\\"__html\\\":\\\"body{color:#000;background:#fff;margin:0}.next-error-h1{border-right:1px solid rgba(0,0,0,.3)}@media (prefers-color-scheme:dark){body{color:#fff;background:#000}.next-error-h1{border-right:1px solid rgba(255,255,255,.3)}}\\\"}}],[\\\"$\\\",\\\"h1\\\",null,{\\\"className\\\":\\\"next-error-h1\\\",\\\"style\\\":{\\\"display\\\":\\\"inline-block\\\",\\\"margin\\\":\\\"0 20px 0 0\\\",\\\"padding\\\":\\\"0 23px 0 0\\\",\\\"fontSize\\\":24,\\\"fontWeight\\\":500,\\\"verticalAlign\\\":\\\"top\\\",\\\"lineHeight\\\":\\\"49px\\\"},\\\"children\\\":\\\"404\\\"}],[\\\"$\\\",\\\"div\\\",null,{\\\"style\\\":{\\\"display\\\":\\\"inline-block\\\"},\\\"children\\\":[\\\"$\\\",\\\"h2\\\",null,{\\\"style\\\":{\\\"fontSize\\\":14,\\\"fontWeight\\\":400,\\\"lineHeight\\\":\\\"49px\\\",\\\"margin\\\":0},\\\"children\\\":\\\"This page could not be found.\\\"}]}]]}]}]],\\\"notFoundStyles\\\":[],\\\"styles\\\":null}]}]]}],null],null],\\\"couldBeIntercepted\\\":false,\\\"initialHead\\\":[false,\\\"$L6\\\"],\\\"globalErrorComponent\\\":\\\"$7\\\",\\\"missingSlots\\\":\\\"$W8\\\"}]]\\n\"])</script><script>self.__next_f.push([1,\"6:[[\\\"$\\\",\\\"meta\\\",\\\"0\\\",{\\\"name\\\":\\\"viewport\\\",\\\"content\\\":\\\"width=device-width, initial-scale=1\\\"}],[\\\"$\\\",\\\"meta\\\",\\\"1\\\",{\\\"charSet\\\":\\\"utf-8\\\"}],[\\\"$\\\",\\\"meta\\\",\\\"2\\\",{\\\"name\\\":\\\"next-size-adjust\\\"}]]\\n2:null\\n\"])</script></body></html>"

      47 |       expect(result).toContain('Client: index.default')
      48 |       expect(result).toContain('Client subpath: subpath.default')
    > 49 |       expect(result).toContain('opt-out-react-version: 18.2.0')
         |                      ^
      50 |     })
      51 |   })
      52 |

      at toContain (e2e/app-dir/app-external/app-external.test.ts:49:22)
      at Object.<anonymous> (e2e/app-dir/app-external/app-external.test.ts:43:5)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| buildDuration | 25s | 23.4s | N/A |
| buildDurationCached | 13.7s | 12s | N/A |
| nodeModulesSize | 360 MB | 359 MB | N/A |
| nextStartRea..uration (ms) | 648ms | 541ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| 139-HASH.js gzip | 31.8 kB | 31.8 kB | N/A |
| 2478adb3-HASH.js gzip | 53.5 kB | 53.5 kB | N/A |
| 4967-HASH.js gzip | 5.1 kB | 5.1 kB | N/A |
| 6701.HASH.js gzip | 168 B | 168 B | ✓ |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 229 B | 225 B | N/A |
| main-HASH.js gzip | 31.6 kB | 31.6 kB | N/A |
| webpack-HASH.js gzip | 1.65 kB | 1.64 kB | N/A |
| Overall change | 45.3 kB | 45.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 193 B | N/A |
| _error-HASH.js gzip | 192 B | 192 B | ✓ |
| amp-HASH.js gzip | 510 B | 510 B | ✓ |
| css-HASH.js gzip | 343 B | 342 B | N/A |
| dynamic-HASH.js gzip | 2.51 kB | 2.52 kB | N/A |
| edge-ssr-HASH.js gzip | 266 B | 265 B | N/A |
| head-HASH.js gzip | 365 B | 362 B | N/A |
| hooks-HASH.js gzip | 391 B | 390 B | N/A |
| image-HASH.js gzip | 4.32 kB | 4.32 kB | N/A |
| index-HASH.js gzip | 268 B | 268 B | ✓ |
| link-HASH.js gzip | 2.69 kB | 2.69 kB | N/A |
| routerDirect..HASH.js gzip | 328 B | 328 B | ✓ |
| script-HASH.js gzip | 393 B | 395 B | N/A |
| withRouter-HASH.js gzip | 323 B | 323 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.73 kB | 1.73 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 484 B | ✓ |
| Overall change | 484 B | 484 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| index.html gzip | 528 B | 528 B | ✓ |
| link.html gzip | 540 B | 541 B | N/A |
| withRouter.html gzip | 523 B | 523 B | ✓ |
| Overall change | 1.05 kB | 1.05 kB | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| edge-ssr.js gzip | 108 kB | 94.7 kB | N/A |
| page.js gzip | 3.04 kB | 3.05 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 622 B | 623 B | N/A |
| middleware-r..fest.js gzip | 155 B | 154 B | N/A |
| middleware.js gzip | 27.9 kB | 27.9 kB | ✓ |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 28.8 kB | 28.8 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 171 kB | 171 kB | ✓ |
| app-page-exp..prod.js gzip | 98.4 kB | 98.4 kB | ✓ |
| app-page-tur..prod.js gzip | 99.9 kB | 99.9 kB | ✓ |
| app-page-tur..prod.js gzip | 94.2 kB | 94.2 kB | ✓ |
| app-page.run...dev.js gzip | 157 kB | 157 kB | ✓ |
| app-page.run..prod.js gzip | 93 kB | 93 kB | ✓ |
| app-route-ex...dev.js gzip | 21.4 kB | 21.4 kB | ✓ |
| app-route-ex..prod.js gzip | 15.2 kB | 15.2 kB | ✓ |
| app-route-tu..prod.js gzip | 15.2 kB | 15.2 kB | ✓ |
| app-route-tu..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route.ru...dev.js gzip | 21.3 kB | 21.3 kB | ✓ |
| app-route.ru..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.82 kB | 9.82 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-turbo...prod.js gzip | 21.4 kB | 21.4 kB | ✓ |
| pages.runtim...dev.js gzip | 22.1 kB | 22.1 kB | ✓ |
| pages.runtim..prod.js gzip | 21.4 kB | 21.4 kB | ✓ |
| server.runti..prod.js gzip | 65.3 kB | 51.6 kB | N/A |
| Overall change | 910 kB | 910 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js revert-64799-sam/fetch-cache/fix-check | Change |
| - | - | - | - |
| 0.pack gzip | 1.61 MB | 1.61 MB | N/A |
| index.pack gzip | 112 kB | 112 kB | ⚠️ +408 B |
| Overall change | 112 kB | 112 kB | ⚠️ +408 B |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>server.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 712f71a1611b95dfa49bccad9ee0277cf8253778

**Review by eps1lon — APPROVED** [review]

File: packages/next/src/server/lib/incremental-cache/fetch-cache.ts [hunk 1]
```diff
@@ -253,22 +233,19 @@ export default class FetchCache implements CacheHandler {
...
 
-        const { data: cached } = parsed
-
         // if new tags were specified, merge those tags to the existing tags
         if (cached.kind === 'FETCH') {
           cached.tags ??= []
           for (const tag of tags ?? []) {
             if (!cached.tags.includes(tag)) {
-              cached.tags.push(tag)
+              cached.tag.push(tag)
```
Comment (samcx) [inline packages/next/src/server/lib/incremental-cache/fetch-cache.ts:248]:
```suggestion
              cached.tags.push(tag)
```

This was part of the typo fix.

**Review by samcx — CHANGES_REQUESTED** [review]

File: packages/next/src/server/lib/incremental-cache/fetch-cache.ts [hunk 2]
```diff
@@ -253,16 +233,13 @@ export default class FetchCache implements CacheHandler {
           throw new Error(`invalid response from cache ${res.status}`)
         }
 
-        const json: IncrementalCacheValue = await res.json()
-        const parsed = zCachedFetchValue.safeParse(json)
+        const cached = await res.json()
```
Comment (ijjk) [inline packages/next/src/server/lib/incremental-cache/fetch-cache.ts:236]:
```suggestion
        const cached: IncrementalCacheValue = await res.json()
```
Let's avoid the any here again this was key part I think

**Review by samcx — APPROVED** [review]

**Pull request merged**